pkg/compact/compact.go

```diff
@@ -278,34 +362,85 @@ func (c *Syncer) GarbageCollect(ctx context.Context) error {
 		if err != nil {
 			return errors.Wrapf(err, "delete block %s from bucket", id)
 		}
+		c.metrics.garbageCollectedBlocks.Inc()
 	}
 	return nil
 }
 
+type groupMetrics struct {
+	cachedBlocksCount float64
+
+	compactionFailures prometheus.Counter
+	compactionDuration prometheus.Histogram
+}
+
+func newGroupMetrics(reg *prometheus.Registry, labels map[string]string, getBlocksFunc func() (ids []ulid.ULID, err error)) *groupMetrics {
+	var m groupMetrics
+
+	groupBlocksCount := prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Name:        "thanos_compact_group_blocks_count",
+		Help:        "Number of currently loaded blocks for group.",
+		ConstLabels: labels,
```

Comment:
Mh, this tends to be rather dangerous I believe since it attaches labels dedicated to the global hierarchy (i.e. attached as target labels) into a metric. This will most likely cause odd conflicts.

Probably better to just use the group hash as a label. I know it's less insightful, but certainly safer.

Reply:
yeah, good point. Sad ):

Reply:
what about `group_labels=labels.String()` and concatenate all external labels? Can we have long label values? (:

Reply:
It would be odd in some way, but not completely weird :) Let's try it